storage/watchable_store_test.go

```diff
@@ -222,3 +240,73 @@ func TestUnsafeAddWatching(t *testing.T) {
 		}
 	}
 }
+
+func TestMapWatchingToEvents(t *testing.T) {
+	var (
+		testN         = 5
+		testKey       = []byte("Foo")
+		testValue     = []byte("Bar")
+		testKeys      = make([][]byte, testN)
+		testValues    = make([][]byte, testN)
+		testWatchings = make([]*watching, testN)
+		testEvents    = make([]storagepb.Event, testN)
+	)
+	for i := 0; i < testN; i++ {
+		k, v := testKey, testValue
+		if i > 0 {
+			k = []byte(fmt.Sprintf("%s_%d", testKey, i))
+			v = []byte(fmt.Sprintf("%s_%d", testValue, i))
+		}
+		testKeys[i] = k
+		testValues[i] = v
+		testWatchings[i] = &watching{
+			key:    k,
+			prefix: true,
+		}
+		testEvents[i] = storagepb.Event{
+			Type: storagepb.PUT,
+			Kv: &storagepb.KeyValue{
+				Key:   testKey,
+				Value: testValue,
+			},
+		}
+	}
+
+	tests := []struct {
+		sync map[string]map[*watching]struct{}
+		evs  []storagepb.Event
+
+		wwe map[*watching][]storagepb.Event
+	}{
+		{
```

Comment:
i do not really like the test case. I think we need multiple test cases for the func, not only one. And we should know why we want to construct the test case in a particular way rather than just put 5 random key and watching in the map.